src/mdx/versionedLinks.js

```diff
@@ -0,0 +1,22 @@
+const visitPromise = import("unist-util-visit");
+const expandVersionedUrl = require("./expandVersionedUrl");
+
+const resolve = async () => {
+  visit = (await visitPromise).visit;
+};
+
+resolve();
```

Comment:
This `import`/`resolve` nonsense irritates me, but I was led here due to the following: 

1. `require` is not supported. And in fact it suggest I use a dynamic `import()` instead:

```
[ERROR] Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/stevenhicks/sjh/dev/camunda/camunda-platform-docs/node_modules/unist-util-visit/index.js from /Users/stevenhicks/sjh/dev/camunda/camunda-platform-docs/src/mdx/versionedLinks.js not supported.
Instead change the require of index.js in /Users/stevenhicks/sjh/dev/camunda/camunda-platform-docs/src/mdx/versionedLinks.js to a dynamic import() which is available in all CommonJS modules.
```

I don't know enough about modules to know why this is, or if there's a way to make it accept `require(...)`. 

2. `await import()` is not supported, no top-level await: 
```
[ERROR] /Users/stevenhicks/sjh/dev/camunda/camunda-platform-docs/src/mdx/versionedLinks.js:1
const visit = await import("unist-util-visit");
              ^^^^^

SyntaxError: await is only valid in async functions and the top level bodies of modules
```

I thought that maybe I could [rename from `.js` to `.mjs` files to signal that they were modules](https://stackoverflow.com/a/65018201), but that didn't work.

I didn't want to spend a lot of time dealing with build issues, so I just accepted this code.

Reply:
Huh, is this d-🦖?

Reply:
I think so, but what's confusing is that the docs for writing custom MDX plugins in docusaurus use `require`. I'm planning to reach out via discord to see if I'm the only one experiencing this.

Reply:
Apparently there is a disclaimer in the docs that is intended to explain this:

> NOTE
> There's recently a trend in the Remark/Rehype ecosystem to migrate to ES Modules, a new JavaScript module system, which Docusaurus doesn't support yet. Please make sure your installed plugin version is CommonJS-compatible before we officially support ESM. Alternatively, you can read about using dynamic import() as a workaround in the tutorial of installing [rehype-katex](https://docusaurus.io/docs/markdown-features/math-equations#upgrading-rehype-katex-beyond-recommended-version).

If we want to use `require` instead of dynamic imports, [we need to install an older version of unist-util-visit](https://discord.com/channels/398180168688074762/867060369087922187/1011369318807765072): 

> You need to use unist-util-visit v2


I'm personally fine with this clunky import/await code so I don't intend on rewriting with that version.